### PR TITLE
Verify that `Manager.from_queryset()` handles invalid argument types

### LINKED ISSUE #905: AssertionError raised when calling Manager.from_queryset()
# Bug report

## What's wrong

When the argument to `models.Manager.from_queryset()` is a type alias, the following `AssertionError` is raised:

```
version: 0.942
Traceback (most recent call last):
  File "mypy/semanal.py", line 5220, in accept
  File "mypy/nodes.py", line 1155, in accept
  File "mypy/semanal.py", line 2091, in visit_assignment_stmt
  File "mypy/semanal.py", line 2377, in apply_dynamic_class_hook
  File "/Users/robpercival/.virtualenvs/kraken-core/lib/python3.8/site-packages/mypy_django_plugin/transformers/managers.py", line 173, in create_new_manager_class_from_from_queryset_method
    assert isinstance(derived_queryset_info, TypeInfo)
AssertionError:
```

If it is a class with a type argument, it instead raises here:

```
version: 0.942
Traceback (most recent call last):
  File "mypy/semanal.py", line 5220, in accept
  File "mypy/nodes.py", line 1155, in accept
  File "mypy/semanal.py", line 2091, in visit_assignment_stmt
  File "mypy/semanal.py", line 2377, in apply_dynamic_class_hook
  File "/Users/robpercival/.virtualenvs/kraken-core/lib/python3.8/site-packages/mypy_django_plugin/transformers/managers.py", line 148, in create_new_manager_class_from_from_queryset_method
    assert isinstance(passed_queryset, NameExpr)
AssertionError:
```

## How is that should be

It should support both type aliases and classes with type arguments.

## System information

- OS: macOS 12.3
- `python` version: 3.8.3
- `django` version: 3.2.10
- `mypy` version: 0.942
- `django-stubs` version: 1.10.1
- `django-stubs-ext` version: 0.4.0

### PR DESCRIPTION
**Pull request by flaeppe**
Test providing some quite odd types/expressions as `Manager.from_queryset()` arguments to see that it won't crash

## Related issues

Closes: #905

### CODE REVIEW

File: tests/typecheck/managers/querysets/test_from_queryset.yml [hunk 1]
```diff
@@ -675,3 +675,36 @@
...
 
               StrCallable = BaseManager.from_queryset(ModelQuerySet, class_name=str(1))
+
+-   case: test_queryset_arg_as_unsupported_expressions
+    main: |
+        from myapp.models import ManagerFromAlias, ManagerFromNonQSGeneric
+        reveal_type(ManagerFromAlias)
+        reveal_type(ManagerFromNonQSGeneric)
+    out: |
+        main:2: note: Revealed type is "Type[django.db.models.manager.Manager[django.db.models.base.Model]]"
```
Comment (sobolevn) [inline tests/typecheck/managers/querysets/test_from_queryset.yml]: Shouldn't it be `Any`? Usually when some type error happens, we are not sure about the result and set it to `Any.from_error`.

Comment (flaeppe) [reply tests/typecheck/managers/querysets/test_from_queryset.yml]: Yeah, probably. I think this just returns the default return type right now.

Comment (flaeppe) [reply tests/typecheck/managers/querysets/test_from_queryset.yml]: Hm, thinking twice about this. Should it really? Our plugin has bailed here and this is just mypy deciding to use the return type from the signature.

So what is happening here shouldn't be anything other than default mypy behaviour.

**Review by intgr — APPROVED** [review]
Makes sense to me. @sobolevn do you want to take another look at this?

File: tests/typecheck/managers/querysets/test_from_queryset.yml [hunk 2]
```diff
@@ -675,3 +675,28 @@
...
+
+        T = TypeVar("T")
+        class NonQSGeneric(Generic[T]):
+            ...
+        reveal_type(Manager[models.Model].from_queryset(NonQSGeneric[int]))
+    out: |
+        main:12: note: Revealed type is "Type[django.db.models.manager.Manager[django.db.models.base.Model]]"
+        main:12: error: Argument 1 to "from_queryset" of "BaseManager" has incompatible type "<typing special form>"; expected "Type[_QuerySet[Model, Model]]"
+        main:17: note: Revealed type is "Type[django.db.models.manager.Manager[django.db.models.base.Model]]"
+        main:17: error: Argument 1 to "from_queryset" of "BaseManager" has incompatible type "Type[NonQSGeneric[Any]]"; expected "Type[_QuerySet[Model, Model]]"
```
Comment (intgr) [inline tests/typecheck/managers/querysets/test_from_queryset.yml]:
```suggestion
        main:12: note: Revealed type is "Type[django.db.models.manager.Manager[django.db.models.base.Model]]"
        main:12: error: Argument 1 to "from_queryset" of "BaseManager" has incompatible type "<typing special form>"; expected "Type[_QuerySet[Model, Model]]"  [arg-type]
        main:17: note: Revealed type is "Type[django.db.models.manager.Manager[django.db.models.base.Model]]"
        main:17: error: Argument 1 to "from_queryset" of "BaseManager" has incompatible type "Type[NonQSGeneric[Any]]"; expected "Type[_QuerySet[Model, Model]]"  [arg-type]
```

**Pull request merged**